plugins/woocommerce-admin/client/products/fields/attribute-control/attribute-control.tsx

```diff
@@ -9,56 +9,76 @@ import {
 	__experimentalSelectControlMenuSlot as SelectControlMenuSlot,
 	Link,
 } from '@woocommerce/components';
-import { recordEvent } from '@woocommerce/tracks';
 import interpolateComponents from '@automattic/interpolate-components';
 import { getAdminLink } from '@woocommerce/settings';
 
 /**
  * Internal dependencies
  */
 import './attribute-field.scss';
-import { AddAttributeModal } from './add-attribute-modal';
 import { EditAttributeModal } from './edit-attribute-modal';
 import { EnhancedProductAttribute } from '~/products/hooks/use-product-attributes';
 import {
+	getAttributeId,
 	getAttributeKey,
 	reorderSortableProductAttributePositions,
 } from './utils';
 import { AttributeEmptyState } from '../attribute-empty-state';
 import {
-	AddAttributeListItem,
 	AttributeListItem,
+	NewAttributeListItem,
 } from '../attribute-list-item';
+import { NewAttributeModal } from './new-attribute-modal';
 
 type AttributeControlProps = {
 	value: ProductAttribute[];
+	onAdd?: ( attribute: EnhancedProductAttribute[] ) => void;
 	onChange: ( value: ProductAttribute[] ) => void;
-	// TODO: should we support an 'any' option to show all attributes?
-	attributeType?: 'regular' | 'for-variations';
+	onEdit?: ( attribute: ProductAttribute ) => void;
+	onRemove?: ( attribute: ProductAttribute ) => void;
+	onRemoveCancel?: ( attribute: ProductAttribute ) => void;
+	onNewModalCancel?: () => void;
+	onNewModalClose?: () => void;
+	onNewModalOpen?: () => void;
+	onEditModalCancel?: ( attribute?: ProductAttribute ) => void;
+	onEditModalClose?: ( attribute?: ProductAttribute ) => void;
+	onEditModalOpen?: ( attribute?: ProductAttribute ) => void;
+	text?: {
```

Comment:
I wonder if instead of `text`, we should use a more descriptive name such as `uiStrings`, or `uiText`? I think I like the grouping of all the UI strings into a single prop though!

Reply:
I like this name much better!  Updated in https://github.com/woocommerce/woocommerce/pull/36563/commits/71a2867510e9b9d18d6af3a8d8f89cd3fb256fed